### PR TITLE
Add `limit_nofile` parameter

### PR DESCRIPTION
**Pull request by jacobmw**
This will allow changing the LimitNOFILE value in the systemd unit file.

### CODE REVIEW

File: templates/unit.erb [hunk 1]
```diff
@@ -34,7 +34,7 @@ Environment='KAFKA_LOG4J_OPTS=<%= @producer_log4j_opts %>'
 Environment='KAFKA_JMX_OPTS=<%= @producer_jmx_opts %>'
 ExecStart=/opt/kafka/bin/kafka-console-producer.sh <% @producer_service_config.sort.each do |k,v| -%><% unless v.to_s.strip.empty? -%>--<%= k -%>=<%= v.is_a?(Array) ? v.join(',') : v %> <% end -%><% end -%> <%= @input %>
 <%- end -%>
-LimitNOFILE=65536
+LimitNOFILE=1000000
```
Comment (alexjfisher) [inline templates/unit.erb]: This should be
```
LimitNOFILE=<% @limit_nofile %>
```
?

File: spec/acceptance/broker_spec.rb [hunk 2]
```diff
@@ -254,6 +254,7 @@ class { 'kafka::broker':
         it { is_expected.to contain "Environment='KAFKA_LOG4J_OPTS=-Dlog4j.configuration=file:/tmp/log4j.properties'" }
         it { is_expected.to contain "Environment='KAFKA_OPTS=-Djava.security.policy=/some/path/my.policy'" }
         it { is_expected.to contain "Environment='LOG_DIR=/some/path/to/logs'" }
+        it { is_expected.to contain "LimitNOFILE='65536'"}
```
Comment (alexjfisher) [inline spec/acceptance/broker_spec.rb]: I'm not sure the acceptance tests are being run automatically by travis.  Do you think you could also add an rspec-puppet test to https://github.com/voxpupuli/puppet-kafka/blob/master/spec/classes/broker_spec.rb ?

File: templates/unit.erb (hunk 1 above)
Comment (jacobmw) [inline templates/unit.erb]: Yes indeed.  That was the whole intention of this pull request ;)

File: spec/acceptance/broker_spec.rb (hunk 2 above)
Comment (jacobmw) [inline spec/acceptance/broker_spec.rb]: Added in additional rspec-puppet tests

File: templates/unit.erb [hunk 3]
```diff
@@ -34,8 +34,9 @@ Environment='KAFKA_LOG4J_OPTS=<%= @producer_log4j_opts %>'
 Environment='KAFKA_JMX_OPTS=<%= @producer_jmx_opts %>'
 ExecStart=/opt/kafka/bin/kafka-console-producer.sh <% @producer_service_config.sort.each do |k,v| -%><% unless v.to_s.strip.empty? -%>--<%= k -%>=<%= v.is_a?(Array) ? v.join(',') : v %> <% end -%><% end -%> <%= @input %>
 <%- end -%>
-LimitNOFILE=65536
-LimitCORE=infinity
+
+LimitNOFILE=<%= @limit_nofile %>
+LimitCORE=infinity2
```
Comment (alexjfisher) [inline templates/unit.erb]: infinity2 ?

File: spec/acceptance/broker_spec.rb [hunk 4]
```diff
@@ -254,6 +254,7 @@ class { 'kafka::broker':
         it { is_expected.to contain "Environment='KAFKA_LOG4J_OPTS=-Dlog4j.configuration=file:/tmp/log4j.properties'" }
         it { is_expected.to contain "Environment='KAFKA_OPTS=-Djava.security.policy=/some/path/my.policy'" }
         it { is_expected.to contain "Environment='LOG_DIR=/some/path/to/logs'" }
+        it { is_expected.to contain "LimitNOFILE='65536'" }
```
Comment (alexjfisher) [inline spec/acceptance/broker_spec.rb]: Is `'65536'` supposed to be in single quotes here?

**Comment by alexjfisher** [comment]
@jacobmw Looking better.  Can you rebase and squash your commits please?

**Comment by jacobmw** [comment]
I can't seem to be able to squash/rebase :(  I'm just going to close out this PR and recreate on a clean branch

**Pull request closed without merging**